### PR TITLE
Automated cherry pick of #23408

### PR DESCRIPTION
**Pull request by janetkuo**
Automated cherry pick of #23408 upstream release 1.2

@zmerlynn @roberthbailey

### CODE REVIEW

File: examples/guestbook/all-in-one/frontend.yaml [hunk 1]
```diff
@@ -16,24 +16,25 @@ spec:
...
-  labels:
-    app: guestbook
-    tier: frontend
+  # labels:
+  #   app: guestbook
+  #   tier: frontend
 spec:
   # this replicas value is default
   # modify it according to your case
   replicas: 3
```
Comment (roberthbailey) [inline examples/guestbook/all-in-one/frontend.yaml]: Is this a merge issue? I think this line should be `replicas: 3`

File: examples/guestbook/all-in-one/guestbook-all-in-one.yaml [hunk 2]
```diff
@@ -16,26 +16,27 @@ spec:
...
-    role: master
-    tier: backend
+  # labels:
+  #   app: redis
+  #   role: master
+  #   tier: backend
 spec:
   # this replicas value is default
   # modify it according to your case
   replicas: 1
```
Comment (roberthbailey) [inline examples/guestbook/all-in-one/guestbook-all-in-one.yaml]: same here, should be `replicas: 1`

File: examples/guestbook/all-in-one/guestbook-all-in-one.yaml [hunk 3]
```diff
@@ -70,26 +71,27 @@ spec:
...
-    role: slave
-    tier: backend
+  # labels:
+  #   app: redis
+  #   role: slave
+  #   tier: backend
 spec:
   # this replicas value is default
   # modify it according to your case
   replicas: 2
```
Comment (roberthbailey) [inline examples/guestbook/all-in-one/guestbook-all-in-one.yaml]: `replicas: 2`

File: examples/guestbook/all-in-one/redis-slave.yaml [hunk 4]
```diff
@@ -15,26 +15,27 @@ spec:
...
-    role: slave
-    tier: backend
+  # labels:
+  #   app: redis
+  #   role: slave
+  #   tier: backend
 spec:
   # this replicas value is default
   # modify it according to your case
   replicas: 2
```
Comment (roberthbailey) [inline examples/guestbook/all-in-one/redis-slave.yaml]: `replicas: 2`

File: pkg/kubectl/cmd/delete_test.go [hunk 5]
```diff
@@ -375,16 +375,14 @@ func TestDeleteMultipleResourcesWithTheSameName(t *testing.T) {
 }
 
 func TestDeleteDirectory(t *testing.T) {
-	_, svc, rc := testData()
+	_, _, rc := testData()
 
 	f, tf, codec := NewAPIFactory()
 	tf.Printer = &testPrinter{}
 	tf.Client = &fake.RESTClient{
```
Comment (roberthbailey) [inline pkg/kubectl/cmd/delete_test.go]: Should be `tf.Client = &fake.RESTClient{`

File: examples/guestbook/all-in-one/frontend.yaml (hunk 1 above)
Comment (janetkuo) [inline examples/guestbook/all-in-one/frontend.yaml]: If you view the file it looks fine https://github.com/janetkuo/kubernetes/blob/automated-cherry-pick-of-%2323408-upstream-release-1.2/examples/guestbook/all-in-one/frontend.yaml

I think it's github rendering issue?

File: examples/guestbook/all-in-one/guestbook-all-in-one.yaml (hunk 2 above)
Comment (janetkuo) [inline examples/guestbook/all-in-one/guestbook-all-in-one.yaml]: ditto https://github.com/janetkuo/kubernetes/blob/automated-cherry-pick-of-%2323408-upstream-release-1.2/examples/guestbook/all-in-one/guestbook-all-in-one.yaml

File: pkg/kubectl/cmd/delete_test.go (hunk 5 above)
Comment (zmerlynn) [inline pkg/kubectl/cmd/delete_test.go]: That's a surreal conversation to follow on email, BTW. :) Yeah, every one of those you pointed out was identical.

**Pull request merged**